### PR TITLE
Trim SKU before product save

### PR DESCRIPTION
**Pull request by birmo**
### Description

There is a problem with spaces before and after attributes value. The problem is very annoying at SKU. It leads to many misunderstandings because for example 'my-sku' is different than 'my-sku ' (look at the space).
I fixed it adding trim function to every product data value.

### Manual testing scenarios

1. Set the SKU with space before or after the right value in admin.
2. Save product.
3. Space on the beginning and on the end of SKU value should disappear.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14245) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=14245) before we can accept your contribution.<br/>

**Comment by ihor-sviziev** [comment]
@dmanners AFAIK you already discussed this issue as part of import export improvements project. Could you please review this PR?

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php [hunk 1]
```diff
@@ -317,6 +318,25 @@ protected function normalize(array $productData)
         return $productData;
     }
 
+    /**
+     * Internal trim data
+     *
+     * @param array $productData
+     * @return array
+     */
+    protected function trimData(array $productData)
```
Comment (orlangur) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php:327]: This is a copy of `normalize` method which is going to be removed: https://github.com/birmo/magento2/blob/2.2-develop/app/code/Magento/Catalog/Controller/Adminhtml/Product/Initialization/Helper.php#L303

Need to be found out in favor of what `normalize` is going to be removed.

**Comment by dmanners** [comment]
Thanks for letting me know @ihor-sviziev .

And thank you for this pull request @birmo currently we already have a pull request in place for this as part of the import/export project. As such I will close this pull request as a duplicate.

https://github.com/magento-engcom/import-export-improvements/pull/102

**Pull request closed without merging**